vembrane/modules/sort.py

```diff
@@ -0,0 +1,359 @@
+        help="Output file, if not specified, output is written to STDOUT.",
+    )
+    parser.add_argument(
+        "--output-fmt",
+        "-O",
+        default="vcf",
+        choices=["vcf", "bcf", "uncompressed-bcf"],
+        help="Output format.",
+    )
+    parser.add_argument(
+        "--preserve-annotation-order",
+        action="store_true",
+        help="If set, annotations are not sorted within the records, but kept in the "
+        "same order as in the input VCF file. If not set (default), annotations "
+        "are sorted within the record according to the given keys if any of the sort "
+        "keys given in the python expression refers to an annotation.",
+    )
+    parser.add_argument(
+        "--chunk-size",
+        type=int,
+        default=1000,
+        help="Number of VCF records to sort in memory. If the VCF file exceeds this "
+        "number of records, external sorting is used.",
+    )
+    add_common_arguments(parser)
+
+
+def is_na(value: Any) -> bool:
+    return (
+        value is NA or value is None or (isinstance(value, float) and math.isnan(value))
+    )
+
+
+T = TypeVar("T")
+
+type NestedSequence[T] = Sequence[T | NestedSequence[T]]
+
+
+# Key wrapper classes for sorting.
+# sorted() uses <, max and min use < and >, so we need to implement both.
+# For maximizing speed, we avoid delegation, saving an additional method call.
+class KeyBase:
+    @classmethod
+    def wrap(cls, key: Any) -> Self | NestedSequence[Self]:
+        if isinstance(key, Iterable) and not isinstance(key, str):
+            # apply recursively to tuples, lists, etc.
+            return tuple(cls.wrap(k) for k in key)
+        else:
+            return cls(key)
+
+    def __init__(self, key: Any):
+        self.key = key
+
+    # rationale for the type ignore below: the code ensures that other is of same type
+    # hence we avoid additional checks here for performance reasons.
+    def __eq__(self, other: Self) -> bool:  # type: ignore
+        if is_na(self.key) and is_na(other.key):
+            return True
+        return self.key == other.key
+
+
+@total_ordering
+class KeyAscending(KeyBase):
+    def __lt__(self, other: Self) -> bool:
+        if is_na(self.key):
+            return False
+        if is_na(other.key):
+            return True
+        return self.key < other.key
+
+    def __gt__(self, other: Self) -> bool:
+        if is_na(self.key):
+            return True
+        if is_na(other.key):
+            return False
+        return self.key > other.key
+
+
+@total_ordering
+class KeyDescending(KeyBase):
+    def __lt__(self, other: Self) -> bool:
+        if is_na(self.key):
+            return True
+        if is_na(other.key):
+            return False
+        return self.key > other.key
+
+    def __gt__(self, other: Self) -> bool:
+        if is_na(self.key):
+            return False
+        if is_na(other.key):
+            return True
+        return self.key < other.key
```

Comment:
does this not mean that NA/None/nan are sorted to the beginning, which is contrary to what the docs state (i.e. NA always at the end)?